tests/Domain/Entity/GenreTest.php

```diff
@@ -47,4 +47,41 @@ public function testRequestingUnfetchedParentThrowsException()
 
         $genre->getParent();
     }
+
+    public function testGetAncestryWithParent()
+    {
+        $parentGenre = new Genre([0], 'parent_id', 'Parent Title', 'parent_url_key', null);
+        $genre = new Genre([0, 1], 'id', 'Title', 'url_key', $parentGenre);
+
+        $ancestry = $genre->getAncestry();
+
+        $this->assertEquals($ancestry, [$genre, $parentGenre]);
+    }
+
+    public function testGetAncestryWithNoParent()
+    {
+        $genre = new Genre([0], 'id', 'Title', 'url_key', null);
+        $ancestry = $genre->getAncestry();
+
+        $this->assertEquals($ancestry, [$genre]);
+    }
+
+    public function testGetUrlKeyHierarchyWithParent()
+    {
+        $parentGenre = new Genre([0], 'parent_id', 'Parent Title', 'parent_url_key', null);
+        $genre = new Genre([0, 1], 'id', 'Title', 'url_key', $parentGenre);
+
+        $urlKey = $genre->getUrlKeyHierarchy();
+
+        $this->assertEquals($urlKey, "parent_url_key/url_key");
```

Comment:
nice tests @mattgreenham!!. Just a comment about standards (coding style in tests)
As the standard in the PHP community we use first the expected value, and later the variable.
Example:
```diff
- $this->assertEquals($urlKey, "parent_url_key/url_key");
+ $this->assertEquals("parent_url_key/url_key", $urlKey);
```

👍

Reply:
Ooh, fixed!